### PR TITLE
[9.x] Add pseudo namespace for tables for `make:model` command 

### PR DESCRIPTION
**Pull request by Sophist-UK**
Laravel makes extensive use of php namespaces to allow packages to co-exist, however the same is not true of SQL tables, with the consequence that the package that creates tables cannot easily be identified and increasing the chances of clashes on table names between packages.

This pull request adds pseudo namespace support for tables to the `make:model` command by pre-pending the path specified on the command line to the table name in the model and migration files. So currently...
* `php artisan make:model Test` uses `test` as the table name
* `php artisan make:model Project\Test` also uses `test` as the table name

This pull request prepends the path to the table name in the migration and model files as follows:
* `php artisan make:model Test` still uses `test` as the table name
* `php artisan make:model Project\SubProj\Test` now uses `project_subproj_test` as the table name

Obviously the `make:model` command only creates stub files, and the user then has the option to change the table names, so this PR should not introduce any real backward compatibility issues, but hopefully this change will encourage package providers to group their tables.

Note: This is my first PR on Laravel so please excuse any mistakes in the coding or submission.

### CODE REVIEW

**Comment by taylorotwell** [comment]
Thanks for your pull request to Laravel!

Unfortunately, I'm going to delay merging this code for now. To preserve our ability to adequately maintain the framework, we need to be very careful regarding the amount of code we include. 

If possible, please consider releasing your code as a package so that the community can still take advantage of your contributions! 

If you feel absolutely certain that this code corrects a bug in the framework, please "**@**" mention me in a follow-up comment with further explanation so that GitHub will send me a notification of your response.

**Comment by Sophist-UK** [comment]
@taylorotwell Taylor

This is the second PR which has been closed this way. This one is not a bug either but IMO this fixes a poor design whereby component developers are **NOT** encouraged to namespace their tables.

Whilst I understand the point you are making, can you point me to some documentation which explains this policy?

Does the Laravel project accept any enhancement PRs? If so, what are the criteria, how do I ascertain whether an idea will be accepted if I develop a PR for it before I waste the effort?

**Comment by dennisprudlo** [comment]
This is a template message for closing PRs. In this case the first paragraph might be the most important one. There's not really a checklist to see what to do so a PR gets merged. You can always start a discussion first about a feature you plan to implement and see what other developers/contributors think about what you plan to PR. In the end Taylor decides what gets merged into his project, but if a PR receives positive feedback from the community it's more likely to get merged.

Me personally, I don't like this part where the `$table` property is always added when creating a model. The laravel way is to let the framework assume what the table name might be. And only if necessary allow users to add this property to define the table name.

**Comment by BrandonSurowiec** [comment]
Laravel has the ability to export the stubs so you can customize them for your own projects: https://laravel.com/docs/8.x/artisan#stub-customization

Granted, in this case you wouldn't be able to fill in the `$table` property, but you could update your stub to append it. 

However, like @dennisprudlo mentioned, Laravel has conventions to make things easier / require less customization. The more you can follow these conventions, the easier your life as a developer will be: https://laravel.com/docs/8.x/eloquent#table-names

So in short, this PR strays from the Laravel conventions, *and you can customize the stubs for your project) which is why it was closed. The "canned" response isn't always clear, I agree. It would be nice to have it mention a specific reason, but it saves Taylor a bunch of time. It is what it is.

**Comment by Sophist-UK** [comment]
> Me personally, I don't like this part where the `$table` property is always added when creating a model. The laravel way is to let the framework assume what the table name might be. And only if necessary allow users to add this property to define the table name.

Yes - I agree - but stubs don't have conditional processing so you can't omit the `$table_name` block if the model is a root object.
